### PR TITLE
Finch+PyData benchmarks

### PR DESCRIPTION
**Pull request by mtsokol**
Issue: https://github.com/willow-ahrens/finch-tensor/issues/24

Finch and PyData asv benchmarks.

Hi @willow-ahrens!
I started working on a list of specific benchmarks that we want to implement and use for measuring improvements for Lazy API and Finch in general:

- sparse matrix-vector (generated with `fsprand`)
  - large 1000x1000, medium 100x100, small 10x10 (calls `sparse.tensordot`)
  - uses CSC, CSR formats
  - `y = Ax + z` (`A` is sparse)
  - eager and compiled mode!

- sparse tensor-tensor
  - uses CSF, GCXS (PyData) formats

- `SDDMM`
  - `A=B∘CD` (`A` and `B` are sparse)
  - eager and compiled mode!

- reductions (sum, prod)

- elementwise (add, mul)

### CODE REVIEW

**Comment by willow-ahrens** [comment]
Tensordot is a fine way to benchmark, though I would almost prefer the `A[:, None]` syntax because it's more granular, so it feels like there are more opportunities to make scheduling mistakes. Also, since finch_tensor and finch are the repos that we want to change, wouldn't we want this suite to live there?

**Comment by willow-ahrens** [comment]
also, I think that large should be 1_000_000 by 1_000_000 with 10_000_000 nnz.

**Comment by mtsokol** [comment]
> Tensordot is a fine way to benchmark, though I would almost prefer the `A[:, None]` syntax because it's more granular

By `None` you mean adding a new dimention at that position? If yes, then this index type isn't supported in `finch-tensor`

> Also, since finch_tensor and finch are the repos that we want to change, wouldn't we want this suite to live there?

Hmm, we plan to add another backend to `pydata/sparse` so we need benchmarks also there. This would mean there are going to be similar benchmarks in `finch-tensor`, `sparse`, and `Finch.jl`.

I'm Ok, with copying these to `finch-tensor` as well, it's just something to remember that they are there.

File: asv.conf.json [hunk 1]
```diff
@@ -3,20 +3,57 @@
...
     "req": {
-      "numba": [],
-      "numpy": []
+      "finch-tensor": [
+        "",
+        null
+      ],
+      "juliacall": [
+        ""
+      ],
```
[the remark this answers is not in the document]
Comment (mtsokol) [reply asv.conf.json]: Right - removed!

File: benchmarks/utils.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import os
+
+import sparse
+
+from asv_runner.benchmarks.mark import SkipNotImplemented
+
+
+def skip_if_finch():
+    if os.environ[sparse._ENV_VAR_NAME] == "Finch":
```
[the remark this answers is not in the document]
Comment (mtsokol) [reply benchmarks/utils.py]: Done!

**Comment by hameerabbasi** [comment]
> I'm Ok, with copying these to finch-tensor as well, it's just something to remember that they are there.

I'd suggest setting up a new repo that can test both libraries instead of copying code, which can get increasingly out of sync. In addition, it might be a nice start to a "sparse extension" to the Array API benchmarks.

**Review by hameerabbasi — APPROVED** [review]
LGTM, thanks @mtsokol!

**Comment by willow-ahrens** [comment]
I don't think we need to copy these over to finch-tensor, and I see why we would want them here. I'll likely add some similar benchmarks in Finch.jl to test the scheduling heuristics.

**Pull request merged**